### ISSUE TITLE
iOS input bug - input fields

### ISSUE THREAD

**Issue by Salan** [ftlabs/fastclick#32]
I've built an application for iOS which implements fastclick. But the app becomes buggy when there are multiple input fields displayed at the same time and the user taps on one of them. E.g: I've a registration form with a few input fields (name, email etc). Tapping a field shifts focus to the wrong input field which quickly becomes very frustating. The keyboard appears at the same time. This generally happens the first time I started googling this issue and found a pretty fresh thread which unfortunately got locked: 

https://github.com/ftlabs/fastclick/issues/17

You say that you cant reproduce the error but I can. Just open http://jsfiddle.net/agH7R/5/ (which was posted in the former thread) and first press "this is link" and then the input field about. The keyboard shows up but focus is moved up towards the CSS code, it doesnt focus on the input field. 

So to sum it up: there is some issues with fastclick ability to determine which input field is clicked, there seem to be problems with setting focus and moving the view to the correct element. Am I the only one who can reproduce this? I've tested it on an iPhone 4 with iOS 5.1.1 and an iPhone 5 with iOS 6. 

Do you have any idea how I can troubleshoot this issue?

**Comment by BrandonLive** [comment]
I'm hitting this same issue with FastClick 1.0.6 on iOS 8.4.1. We have four select controls stacked vertically. If you tap the third one, the page is moved up by the OS select control, and then focus jumps unexpectedly to the 4th select control. If you tap the 4th one, the spinner comes up and then immediately disappears as a click is registered outside the select control.

**Comment by bnek** [comment]
Has anyone yet found a suitable workaround for this problem?

**Comment by powermikee** [comment]
Had the same issue. Only way I found to solve it was to put the "needsclick" class on each input in that form.

**Comment by Taxel** [comment]
I am having the very same problem on iOS with input type="range". I applied the "needsclick" class but that did not help. 
Any ideas how I can get it to work? 
@powermikee did you encounter "range" and successfully fix it?

**Comment by hgroupetech** [comment]
This is how I solved the problem :
`$(function() {
	FastClick.attach('btn, a, button');
});`
